flygym/examples/locomotion/controller_comparison.py

```diff
@@ -0,0 +1,695 @@
+import subprocess
+from itertools import chain, product
+from pathlib import Path
+from typing import Dict, List
+
+import imageio
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import seaborn as sns
+from dm_control.rl.control import PhysicsError
+from joblib import Parallel, delayed
+from matplotlib.image import AxesImage
+from scipy.interpolate import interp1d
+from scipy.stats import mannwhitneyu
+from tqdm import tqdm
+
+from flygym import Camera, SingleFlySimulation
+from flygym.arena import BlocksTerrain, FlatTerrain, GappedTerrain, MixedTerrain
+from flygym.examples.locomotion import CPGNetwork, PreprogrammedSteps
+from flygym.examples.locomotion.rule_based_controller import (
+    RuleBasedController,
+    construct_rules_graph,
+)
+from flygym.preprogrammed import get_cpg_biases
+
+from colored_fly import ColoredFly as Fly
```

Comment:
Actually, what does this do? Is `colored_fly` old code that we can ignore now? @tkclam

Reply:
@sibocw I've just pushed the missing `colored_fly.py` to `add-controller-comparison`. `ColoredFly` solves the issue of body segments not colored accurately due to "textures are applied in GL_MODULATE mode, meaning that the texture color and the color specified here are multiplied component-wise" as mentioned in https://mujoco.readthedocs.io/en/stable/XMLreference.html#asset-material-rgba.